SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialReceiver.h

```diff
@@ -40,50 +41,43 @@ struct PendingAddComponentWrapper
 	TUniquePtr<SpatialGDK::DynamicComponent> Data;
 };
 
-DECLARE_DELEGATE_OneParam(EntityQueryDelegate, const Worker_EntityQueryResponseOp&);
-DECLARE_DELEGATE_OneParam(ReserveEntityIDsDelegate, const Worker_ReserveEntityIdsResponseOp&);
-DECLARE_DELEGATE_OneParam(CreateEntityDelegate, const Worker_CreateEntityResponseOp&);
-
-DECLARE_MULTICAST_DELEGATE_OneParam(FOnEntityAddedDelegate, const Worker_EntityId);
-DECLARE_MULTICAST_DELEGATE_OneParam(FOnEntityRemovedDelegate, const Worker_EntityId);
-
 UCLASS()
-class USpatialReceiver : public UObject
+class USpatialReceiver : public UObject, public SpatialOSDispatcherInterface
 {
 	GENERATED_BODY()
 
 public:
 	void Init(USpatialNetDriver* NetDriver, FTimerManager* InTimerManager, SpatialGDK::SpatialRPCService* InRPCService);
 
 	// Dispatcher Calls
-	void OnCriticalSection(bool InCriticalSection);
-	void OnAddEntity(const Worker_AddEntityOp& Op);
-	void OnAddComponent(const Worker_AddComponentOp& Op);
-	void OnRemoveEntity(const Worker_RemoveEntityOp& Op);
-	void OnRemoveComponent(const Worker_RemoveComponentOp& Op);
-	void FlushRemoveComponentOps();
-	void RemoveComponentOpsForEntity(Worker_EntityId EntityId);
-	void OnAuthorityChange(const Worker_AuthorityChangeOp& Op);
+	virtual void OnCriticalSection(bool InCriticalSection) override;
```

Comment:
Any thoughts on the performance implications of making all these functions virtual?

Reply:
I'm not too worried actually. Given that all of them are calls which will result in network traffic, I would expect that the network latency would overshadow any performance hit from the virtualization. Of course, if they start to show up in a profile we'd need to reconsider.

Reply:
These are responding to network traffic no? So potentially a virtual function call per spatial op? We should add some profiling in the dispatcher to track it at least.

Reply:
Yes, these in particular are on the inbound side, but we're also doing it on the outbound side. The point was more that the network communication would be orders of magnitue longer than the virtual function overhead. I'm totally happy to instrument it though. Do we have docs on the instrumentation we're adding for the latency tracking? Or a PR you could point me towards?

Reply:
https://improbableio.atlassian.net/browse/UNR-322